### PR TITLE
MRG: Refactor/bids compatible with MNE-BIDS/MNE-Python, BidsRun Func, Example Setup and Reading

### PR DESCRIPTION
**Pull request by adam2392**
Addresses: #6 
Improves upon: #4 #2 

**Summary of changes:**
MNE-BIDs and MNE-Python do some things very well, so refactored to adhere to their way of doing things. You'll see that I stripped basically the necessity for us to pass in explicit BIDs parameters. Rather, we can pass in bids_fname, or bids_basename, and bids_root, which is sufficient to get ALL the data we need using internal functions of mne-bids like `func:_parse_bids_filename`, `func:_handle_kind`.

See `read_eeg_from_bids.py` for example on how to do things and get it working in the same framework. Caution: possible bugs.

**Misc comments:**
Note that BidsPatient was just completely ignored now. BidsRun extends the existing nice reading functionality of MNE-Bids to allow adaptive changing of *run* datasets in edf/fif format. This is the main advantage of ours over MNE-BIDS. Overall their writing and other stuff is really good, so the adherence to their code base is strict, but will buy us a lot of bucks. 

The various pipeline I did to set up the BIDS format is hacked together... but we should be able to clean it up and refactor when necessary. This is basically the key to get our version on EZTrack working. Once we "setup", ideally we can either use MNE-BIDS, or our package for directly reading. When we need to programatically modify metadata, we can use BidsRun, which is shown in the corresponding unit tests: `test_basebids.py`, `test_bidsrun.py`.

**TODO:**
1. in future PR, we should get bidspatient working.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,83 +21,38 @@ EEGio is intended to be a lightweight wrapper for easily analyzing large batches
...
-
-2. loaders/
-This defines code that links together different parts of the eegio API to allow easy data pipelines to be setup.
-
-3. writers/
-This defines writers of data to save into hdf, npy, mat files.
+For more info, see tutorials and documentation.
 
 # Contributing
 We welcome contributions from anyone. Our [issues](https://github.com/adam2392/eegio/issues) page is full of places we could use help!
```
Comment (pmyers16) [inline README.md]:
```suggestion
We welcome contributions from anyone. Our [issues](https://github.com/adam2392/eegio/issues) page is a great place for suggestions.
```

Comment (pmyers16) [inline README.md]:
```suggestion
We welcome contributions from anyone. Our [issues](https://github.com/adam2392/eegio/issues) page is a great place for suggestions!
```

File: eegio/base/utils/bids_helper.py [hunk 2]
```diff
@@ -106,12 +113,14 @@ class BidsConverter:
...
         bids_basename,
+        acquisition,
         excluded_contacts=None,
         eog_contacts=None,
         misc_contacts=None,
         overwrite=False,
+        line_freq=None,
         use_preprocess_pipeline=True,
     ):
         """
```
Comment (pmyers16) [inline eegio/base/utils/bids_helper.py]:
```suggestion
        """
        # TODO: Clean up how write_raw_bids is called - eliminate redundant writing/reading using temporaryDirectory
```

File: eegio/base/utils/bids_helper.py [hunk 3]
```diff
@@ -334,111 +372,104 @@ def add_notes_to_channels(bidsrun):
...
-    verbose: bool
-        Whether to print the data to stdout
-
-    """
-    if os.path.exists(fname) and not overwrite:
-        raise FileExistsError(
-            '"%s" already exists. Please set '  # noqa: F821
-            "overwrite to True." % fname
+if __name__ == "__main__":
+    DATADIR = "/Users/Adam2392/Documents/eegio/"
```
Comment (pmyers16) [inline eegio/base/utils/bids_helper.py]: remove references to local paths

File: eegio/base/utils/bids_helper.py [hunk 4]
```diff
@@ -334,111 +372,104 @@ def add_notes_to_channels(bidsrun):
...
-        Whether to print the data to stdout
-
-    """
-    if os.path.exists(fname) and not overwrite:
-        raise FileExistsError(
-            '"%s" already exists. Please set '  # noqa: F821
-            "overwrite to True." % fname
+        print("Loading filepath: ", edf_fpath)
+        print("Writing to bidsroot: ", bids_root)
+        print("Bids basenmae; ", bids_basename)
```
Comment (pmyers16) [inline eegio/base/utils/bids_helper.py]:
```suggestion
        print("Bids basename; ", bids_basename)
```

File: eegio/base/utils/bids_helper.py [hunk 5]
```diff
@@ -334,111 +372,104 @@ def add_notes_to_channels(bidsrun):
...
+        # call bidsbuilder pipeline
+        bids_path = BidsConverter.convert_to_bids(
+            edf_fpath=edf_fpath,
+            bids_root=bids_root,
+            bids_basename=bids_basename,
+            acquisition=modality,
+            line_freq=60,
```
Comment (pmyers16) [inline eegio/base/utils/bids_helper.py]: should line_freq be 60.0 for consistency?

File: eegio/base/utils/bids_helper.py [hunk 6]
```diff
@@ -334,111 +372,104 @@ def add_notes_to_channels(bidsrun):
...
+            # add a bids run
+            bids_basename = make_bids_basename(
+                subject=test_subjectid,
+                session=test_sessionid,
+                run=test_runid,
+                acquisition=modality,
+            )
+            print("Loading filepath: ", edf_fpath)
+            print("Writing to bidsroot: ", bids_root)
+            print("Bids basenmae; ", bids_basename)
```
Comment (pmyers16) [inline eegio/base/utils/bids_helper.py]:
```suggestion
            print("Bids basename; ", bids_basename)
```

File: eegio/base/utils/scrubber.py [hunk 7]
```diff
@@ -11,19 +11,21 @@
 class ChannelScrub:
     @classmethod
     def channel_text_scrub(cls, raw: mne.io.BaseRaw):
+        """TODO: ACCOUNT FOR - BEING IN BIPOLAR MONTAGE CHANNEL LABELS. """
+
         def _reformatchanlabel(label):
             """
```
Comment (pmyers16) [inline eegio/base/utils/scrubber.py]: convert to numpy docstring

File: eegio/loaders/bids/basebids.py [hunk 8]
```diff
@@ -6,20 +6,18 @@
...
+from mne_bids.utils import _write_json
 from mne_bids.utils import print_dir_tree
 
-from eegio.base.utils.bids_helper import write_json
-
 
 class BaseBids(object):
     """
-    An instantiator class of the bids format that any BIDS derivative would use.
+    An instantiator class of the bids format that extends usage of BIDS
```
Comment (pmyers16) [inline eegio/loaders/bids/basebids.py]: In general, the summary statement for the docstring format needs to be one line and end with a period. You can add a line with more detail after.

File: examples/read_eeg_from_bids.py [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+raw = bidsrun.load_data()
+
+print(raw)
+
+###############################################################################
+# Step 4: Run BIDS-Validate
+# ------------------------------------------------------------
+# Now we have written our BIDS directory.
+# save a fif copy and reload it
+# TODO: figure out why this isn't returning valid, when it is valid.
```
Comment (pmyers16) [inline examples/read_eeg_from_bids.py]: Call all of the checks independently to figure out which one is not passing:

- is_top_level
- is_associated_data
- is_session_level
- is_subject_level
- is_phenotypic
- is_file

File: tests/api/test_bids_setup.py [hunk 10]
```diff
@@ -28,44 +30,37 @@ def test_bids_setup(self, bids_dir, edf_fpath):
...
         test_sessionid = "seizure"
         test_runid = "01"
-        # bids_dir = os.path.join(os.getcwd(), "./data/bids_layout/")
-        BidsBuilder.make_bids_folder_struct(
-            datadir=bids_dir,
-            subject=test_subjectid,
-            kind=modality,
-            session_id=test_sessionid,
-        )
+        line_freq = 60
```
Comment (pmyers16) [inline tests/api/test_bids_setup.py]: line_freq = 60.0 for consistency?

**Comment by adam2392** [comment]
Updated pydocstyle adherence, except for "*derivatives/*", which I think we can handle later on. Lmk what you think.

**Pull request merged**